### PR TITLE
[docs] Improve documentation for fontTools.ttLib.ttFont

### PR DESCRIPTION
**Pull request by simoncozens**
- Allow cross-references in docs
- Point users more clearly to ttFont
- Remove old example
- Typos
- Improve documentation for ttFont module

### CODE REVIEW

**Comment by chrissimpkins** [comment]
It looks like the RTD configuration now supports builds on PR's.  Helpful to proof doc updates?

**Comment by simoncozens** [comment]
I'd strongly recommend whoever has admin access to RTD (@anthrotype?) to enable this. My next documentation PR is going to be significant.

**Comment by anthrotype** [comment]
I just enabled that, hope it works. Thanks Simon, I'll take a look soon

**Comment by belluzj** [comment]
@anthrotype I'm interested in the RTD deploy on PR thing, how do I enable it on the designspace PR #2436?

**Comment by anthrotype** [comment]
Honestly I have no idea, I simply followed the above linked instructions to enable the feature. I never used RTD myself

**Comment by chrissimpkins** [comment]
Do you have admin access on the repo in RTD Jany? If you do I can show you where to flip the switch.  It will show up as a new build in the GH CI with a link to the PR specific doc build.

**Comment by anthrotype** [comment]
As I said, I have already flipped that switch. Maybe it'll only show up on new PRs

**Comment by chrissimpkins** [comment]
apologies, was reading this on my phone and was under the assumption that this was for a different repo.

**Comment by simoncozens** [comment]
It doesn’t seem to be working - didn’t show up on #2446 which was created after the config change.

**Comment by chrissimpkins** [comment]
Webhook does not seem to be working for some reason. There must be a RTD configuration issue on the fonttools settings side.  You might need to approve the RTD OAuth app.

@anthrotype I am getting this information on https://readthedocs.org/dashboard/fonttools/integrations/118984/

> The project fonttools doesn't have a valid webhook set up, commits won't trigger new builds for this project. See the [project integrations](https://readthedocs.org/dashboard/fonttools/integrations/) for more information.

and I don't have access to modify webhooks here.

**Comment by anthrotype** [comment]
I went to that page and clicked Resync Webhook, and now I  no longer seeing that message about not having a valid hook set up. Please try again and let me know if it is working now. Also I think a user named "simpkins" is already an admin and co-maintainer on the RTD fontools account.

**Comment by chrissimpkins** [comment]
Yes I have admin access on the RTD account but not on the fonttools repo.  It wouldn't let me re-sync the webhook.  I think this should fix it.

File: Lib/fontTools/ttLib/__init__.py [hunk 1]
```diff
@@ -1,45 +1,4 @@
-"""fontTools.ttLib -- a package for dealing with TrueType fonts.
-
-This package offers translators to convert TrueType fonts to Python
-objects and vice versa, and additionally from Python to TTX (an XML-based
-text format) and vice versa.
-
-Example interactive session:
-
-Python 1.5.2c1 (#43, Mar  9 1999, 13:06:43)  [CW PPC w/GUSI w/MSL]
```
Comment (anthrotype) [inline Lib/fontTools/ttLib/__init__.py:9]: Python 1.5.2c1, March 9 1999... 🥲

File: Lib/fontTools/ttLib/ttFont.py [hunk 2]
```diff
@@ -138,9 +151,13 @@ def close(self):
...
 	def save(self, file, reorderTables=True):
-		"""Save the font to disk. Similarly to the constructor,
-		the 'file' argument can be either a pathname or a writable
-		file object.
+		"""Save the font to disk.
+
+		Args:
+			file: Similarly to the constructor, can be either a pathname or a writable
+				file object.
+			reorderTables (bool): If true (the default), reorder the tables, sorting
```
Comment (anthrotype) [inline Lib/fontTools/ttLib/ttFont.py:159]: reorderTables is an Optional[bool]. When it's set to None (e.g. the subsetter does that), it means write tables in whatever order is most convenient (mostly due to inter-table dependencies at compile time) without caring about either the canonical order or the original font order. This should be the fastest mode

File: Lib/fontTools/ttLib/ttFont.py [hunk 3]
```diff
@@ -320,11 +337,15 @@ def importXML(self, fileOrPath, quiet=None):
 		reader.read()
 
 	def isLoaded(self, tag):
-		"""Return true if the table identified by 'tag' has been
+		"""Return true if the table identified by ``tag`` has been
 		decompiled and loaded into memory."""
 		return tag in self.tables
 
 	def has_key(self, tag):
```
Comment (anthrotype) [inline Lib/fontTools/ttLib/ttFont.py:344]: we should deprecate has_key in favour of `__contains__` at some point

**Review by anthrotype — APPROVED** [review]

**Pull request merged**